### DISCUSSION TITLE
Simulating magnetic anomalies

### DISCUSSION THREAD

**Discussion by SirDshimshi** [fatiando/community#100] [Q&A] [answered]
Hello!
For a project I'm working on i need to calculate the magnetic field intensity at certain points in a cartesian coordinate system, produced by 3d right prisms with specified magnetic moment.

This is something I expected would be possible with harmonica (like it is possible for gravity calculations). It was possible in the deprecated version of fatiando 0.5 with use of the function [fatiando.gravmag.prism.tf()](https://legacy.fatiando.org/api/gravmag.prism.html#fatiando.gravmag.prism.tf)   .
Is this a feature planned for future version of harmonica?

Anyway, does anyone know of any python module that could accomplished this?

**Comment by aguspesce** [comment]
Hello @SirDshimshi!

All the forward models were removed from Harmonica and now live in [Choclo](https://www.fatiando.org/choclo/latest/). 
It is possible that you found the way to calculate the magnetic field using [Choclo](https://www.fatiando.org/choclo/latest/api/index.html). 
This library is optimized for running geophysical forward and inverse models and contains efficient forward modeling and kernel functions. It may prove to be a valuable tool in the calculation of magnetic fields.

**Comment by santisoler** [comment, marked as answer]
Hi @SirDshimshi! Thanks for opening this question.

We are working on the implementation of magnetic forward modelling within Harmonica (you can check it out in https://github.com/fatiando/harmonica/pull/369). As Agustina pointed out, all the underlying code for running the forward models will leave in Choclo in the future: it offers the basics for forward modelling a single source on a single data points. Harmonica will still offer the interface to easily forward model large systems.

Until the forward models arrive to Harmonica, you can already use the ones in Choclo.

Hope it helps!
